internal/jobobject/jobobject.go

```diff
@@ -195,9 +195,13 @@ func Open(ctx context.Context, options *Options) (_ *JobObject, err error) {
 		handle: jobHandle,
 	}
 
+	if job.isOpenedJobSilo() {
```

Comment:
Given we are in the middle of constructing the `job` object here, might be best to avoid calling a method on the object, and instead just call something like `isJobSilo(jobHandle)`. I think methods like `(*JobObject).isOpenedJobSilo` should be able to assume they are operating on a "fully constructed" object.

Reply:
Sure good idea

Reply:
(and agree, this does feel a bit off)

Reply:
@kevpar Done